include/swift/AST/ExtInfo.h

```diff
@@ -517,19 +504,30 @@ class SILExtInfoBuilder {
   SILExtInfoBuilder(unsigned bits, ClangTypeInfo clangTypeInfo)
       : bits(bits), clangTypeInfo(clangTypeInfo) {}
 
+  static constexpr unsigned makeBits(Representation rep, bool isPseudogeneric,
+                                     bool isNoEscape,
+                                     DifferentiabilityKind diffKind) {
+    return ((unsigned)rep) | (isPseudogeneric ? PseudogenericMask : 0) |
+           (isNoEscape ? NoEscapeMask : 0) |
+           (((unsigned)diffKind << DifferentiabilityMaskOffset) &
```

Comment:
Assert that diffKind fits inside DifferentiabilityMask?

Reply:
Hmm, should we do this consistently for all the fields? For example, `representation` takes two bits. I imagine we could have a `Last = 3` case there (and a `Last = 2` case for `DifferentiabilityKind`) and then have a `static_assert(std::bit_width(DifferentiabilityKind::Last) == std::bit_width(DifferentiabilityMask))`.

Reply:
(Leaving this open for discussion. Will fix this in a follow-up PR.)